edweiss-app/components/notifs/NotifDisplay.tsx

```diff
@@ -0,0 +1,259 @@
+    like: 'heart-outline',                  // others:  heart-sharp / thumbs-up-outline
+    follow: 'person-add-outline',
+};
+
+const iconsColor = {
+    message: 'color11',
+    quiz: 'color5',
+    submission: 'color4',
+    grade: 'color17',
+    announcement: 'color1',
+    event: 'color9',
+    meeting: 'color12',
+    group: 'color15',
+    post: 'color13',
+    comment: 'color7',
+    like: 'color8',
+    follow: 'color3',
+};
+
+const iconsColorBackground = {
+    message: "#79B8FE", //color11
+    quiz: "#FFDF5E", //color5
+    submission: "#FFBB5B", //color4
+    grade: "#451F48", //color17
+    announcement: "#651510", //color1
+    event: "#71D3BC", //color9
+    meeting: "#4F6CFD", //color12
+    group: "#844D92", //color15
+    post: "#8F8ED7", //color13
+    comment: "#608E55", //color7
+    like: "#6AC979", //color8
+    follow: "#E97F78",  //color3
+};
+
+// Tests Tags
+export const testIDs = {
+    notifIcon: 'notif-icon',
+    notifDate: 'notif-date',
+    notifTitle: 'notif-title',
+    notifMessage: 'notif-message',
+    swipeView: 'swipe-view',
+};
+
+type dateSection = 'today' | 'thisWeek' | 'thisMonth' | 'thisYear' | 'older';
+
+
+
+const NotifDisplay: ReactComponent<{ item: NotifList.Notif, id: string, dateSection: dateSection, index: number; }> = ({ item, id, dateSection, index }) => {
+
+    // Define swipeableRefs
+    const swipeableRefs = useRef<(Swipeable | null)[]>([]);
+
+    // Render left actions on swipe
+    const renderLeftActions = () => (
+        <TTouchableOpacity onPress={() => {
+            if (item.read) {
+                markAsUnreadAction();
+            } else {
+                markAsReadAction();
+                // const date: Timestamp = Timestamp.fromDate(new Date(1730697600000));
+                // pushNotifAction('meeting', 'Meeting', 'Test Add Notif', false, null, OurTime.toDate(date).toISOString());
+            }
+            swipeableRefs.current[index]?.close();
+        }}>
+            <TView justifyContent='center' alignItems='flex-end' py={20} backgroundColor='blue' testID={testIDs.swipeView}>
+                <TText size={16} bold={true} px={12} color='constantWhite'>
+                    {item.read ? t(`notifications:unread`) : t(`notifications:read`)}
+                </TText>
+            </TView>
+        </TTouchableOpacity>
+
+    );
+
+    const renderRightActions = () => (
+        <TTouchableOpacity onPress={() => {
+            deleteNotifAction();
+            swipeableRefs.current[index]?.close();
+        }}>
+            <TView justifyContent='center' alignItems='flex-end' py={20} backgroundColor='cherry' testID={testIDs.swipeView}>
+                <TText size={16} bold={true} px={12} color='constantWhite'>
+                    {t(`notifications:delete`)}
+                </TText>
+            </TView>
+        </TTouchableOpacity>
+    );
+
+
+    // Fonction pour convertir le code hex en RGB
+    const hexToRgb = (hex: string) => {
```

Comment:
You can add this in a units file. If anyone wants to reuse them.

Reply:
Done ✅